File: K47 User Manual/5_magnetic_sensors/27_hybridReedSwitch/Code/Python/hybridReedSwitch.py
#!/usr/bin/env python
import RPi.GPIO as GPIO
import time

Reed_AO_PIN = 11
LedPin = 16


def init():
	GPIO.setmode(GPIO.BOARD)	
	GPIO.setup(Reed_AO_PIN, GPIO.IN)
	GPIO.setup(LedPin, GPIO.OUT)
	
def loop():	
    while True:
        global digitalVal
        digitalVal = GPIO.input(Reed_AO_PIN)
        if(digitalVal == 0):
			print "Magnet detected!"
			GPIO.output(LedPin, GPIO.HIGH)
			time.sleep(0.2)
        else:
            GPIO.output(LedPin, GPIO.LOW)

if __name__ == '__main__':
	init()
	try:
		loop()
	except KeyboardInterrupt: 
		print 'The end !'

File: K47 User Manual/5_magnetic_sensors/27_hybridReedSwitch/Code/C/hybridReedSwitch.c
#include <wiringPi.h>
#include <stdio.h>
#include <string.h>
#include <errno.h>
#include <stdlib.h>



#define  Reed_AO_Pin   0
#define  LedPin		   4


typedef unsigned char uchar;
typedef unsigned int uint;



int main(void)
{
	uchar digitalVal = 1;
	if(wiringPiSetup() == -1){ //when initialize wiring failed,print messageto screen
		printf("setup wiringPi failed !");
		return 1; 
	}
    
	pinMode(Reed_AO_Pin, INPUT);
	pinMode(LedPin, OUTPUT);

	while(1){
		if((digitalVal = digitalRead(Reed_AO_Pin)) == 0)
		{
			printf("Magnet detected!\n");
			digitalWrite(LedPin, HIGH);
			delay(200);
		}
		else
		{
			digitalWrite(LedPin, LOW);
		}
	}

	return 0;
}

